starter-project-backend/src/controllers/article.ts

```diff
@@ -0,0 +1,142 @@
+import {Response, Request, NextFunction} from "express"
+import Article  from "../models/article"
+
+async function getAllArticle(req:Request, res:Response, next:NextFunction) 
+{
+  try{
+    let articles = await Article.find({}) 
+    res.status(200).json({data: articles})
+  }catch(e)
+  {
+    res.status(500).send("server error")
+  }
+
+}
+
+async function getArticleById( req:Request,res:Response, next:NextFunction) {
+
+
+  if(!req.params.id) {
+    res.status(400).send("bad request")
+    return;
+  }
+
+
+  try{
+
+    let article = await Article.findOne({id: req.params.id}) 
+
```

Comment:
change let to const  
instead of uising findOne it's better to use findById to find objects by Id based on the documentation of mongoose.

Reply:
fixed